reliability-engineering/terraform/modules/concourse-monitoring/files/dashboards/concourse-slo.json

```diff
@@ -0,0 +1,417 @@
+{
+  "annotations": {
+    "list": [
+      {
+        "builtIn": 1,
+        "datasource": "-- Grafana --",
+        "enable": true,
+        "hide": true,
+        "iconColor": "rgba(0, 211, 255, 1)",
+        "name": "Annotations & Alerts",
+        "type": "dashboard"
+      }
+    ]
+  },
+  "description": "service level objectives",
+  "editable": true,
+  "gnetId": null,
+  "graphTooltip": 0,
+  "links": [],
+  "panels": [
+    {
+      "datasource": "CloudWatch",
+      "description": "",
+      "fieldConfig": {
+        "defaults": {
+          "custom": {},
+          "mappings": [
+            {
+              "from": "",
+              "id": 0,
+              "operator": "",
+              "text": "",
+              "to": "",
+              "type": 1,
+              "value": ""
+            }
+          ],
+          "thresholds": {
+            "mode": "percentage",
+            "steps": [
+              {
+                "color": "dark-red",
+                "value": null
+              },
+              {
+                "color": "dark-yellow",
+                "value": 99
+              },
+              {
+                "color": "dark-green",
+                "value": 99.9
+              }
+            ]
+          },
+          "unit": "percentunit"
+        },
+        "overrides": []
+      },
+      "gridPos": {
+        "h": 8,
+        "w": 6,
+        "x": 0,
+        "y": 0
+      },
+      "id": 6,
+      "interval": "",
+      "options": {
+        "colorMode": "value",
+        "graphMode": "area",
+        "justifyMode": "auto",
+        "orientation": "auto",
+        "reduceOptions": {
+          "calcs": [
+            "mean"
+          ],
+          "fields": "/^errorRatio$/",
+          "values": false
+        }
+      },
+      "pluginVersion": "7.0.5",
+      "targets": [
+        {
+          "alias": "",
+          "apiMode": "Metrics",
+          "dimensions": {
+            "LoadBalancer": "app/cd-concourse-web/ba5029fcd9b9cde2"
+          },
+          "expression": "",
+          "hide": true,
+          "id": "status5xx",
+          "matchExact": true,
+          "metricName": "HTTPCode_ELB_5XX_Count",
+          "namespace": "AWS/ApplicationELB",
+          "period": "300",
+          "refId": "A",
+          "region": "eu-west-2",
+          "statistics": [
+            "Sum"
+          ]
+        },
+        {
+          "alias": "",
+          "dimensions": {
+            "LoadBalancer": "app/cd-concourse-web/ba5029fcd9b9cde2"
+          },
+          "expression": "",
+          "hide": true,
+          "id": "status4xx",
+          "matchExact": true,
+          "metricName": "HTTPCode_ELB_4XX_Count",
```

Comment:
This counts 4xxs that originate at the load balancer (so 404s from concourse itself would not be counted).  I don't think this is a useful metric to track at all - I think 404s from the ELB will be either fixed responses (based on a listener rule) or some sort of 400 based on invalid requests.  In any case, a 400 from the ELB is not a concourse error, nor is it an interesting request, so I don't think it should be part of the numerator or denominator of our SLI.

Reply:
done ... dropped 4xx from the availability metric ... I agree they add nothing interesting to the result